### PR TITLE
Razan week 2

### PR DESCRIPTION
**Pull request by roza8765**
# Your Details

Your Name:
Your City:
Your Slack Name:

# Homework Details

Module:
Week:

### CODE REVIEW

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 1]
```diff
@@ -11,7 +11,7 @@
 //    Only make edits inside the function
 
 function mood() {
-  let isHappy = true;
+  let isHappy = false;
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/1-fix-functions.js:14]: good

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 2]
```diff
@@ -26,9 +26,9 @@ function mood() {
 
 function greaterThan10() {
   const num = 10;
-  const isBigEnough;
+  const isBigEnough = "Boolean";
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/1-fix-functions.js:29]: sorry for this being not clear. So usually when you read a variable name that comes as a question like , isItBig, isInfoRight, usually means it's a boolean.
Your work is correct. To follow the above-mentioned practice
```isBigEnough=num>=10```

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ function greaterThan10() {
 
 function sortArray() {
   const letters = ["a", "n", "c", "e", "z", "f"];
-  let sortedLetters;
+  let sortedLetters = letters.sort();
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/1-fix-functions.js:44]: perfect

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 4]
```diff
@@ -52,7 +52,9 @@ function sortArray() {
 
 function first5() {
   let numbers = [1, 2, 3, 4, 5, 6, 7, 8];
-  let sliced;
+
+  let sliced = numbers.splice(0, 5);
+
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/1-fix-functions.js:57]: how about using slice instead :)

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 5]
```diff
@@ -65,7 +67,7 @@ function first5() {
 
 function get3rdIndex(arr) {
   let index = 3;
-  let element;
+  let element = arr[3];
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/1-fix-functions.js:70]: works, but you can use 
arr[index]

File: week-2/Homework/mandatory/2-function-creation.js [hunk 6]
```diff
@@ -5,7 +5,17 @@ Write a function that:
...
 - makes the string all lowercase
 */
-function tidyUpString(strArr) {}
+
+// let strArr = [" /Sanyia ", " Michael ", "AnTHonY ", "   Tim   "];
+
+function tidyUpString(strArr) {
+  let newArray = [];
+  for (let i = 0; i < strArr.length; i++) {
+    newArray.push(strArr[i].trim().toLowerCase().replace(/\//g, ""));
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/2-function-creation.js:14]: works fine
but can you elaborate what //\//g does?
what replace does as well

File: week-2/Homework/mandatory/2-function-creation.js [hunk 7]
```diff
@@ -15,18 +25,30 @@ Complete the function to check if the variable `num` satisfies the following req
 Tip: use logical operators
 */
 
-function validate(num) {}
+function validate(num) {
+
+  if (num === num && num % 2 === 0 && num <= 100) {
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/2-function-creation.js:30]: what is num===num?
you want to check if it's a number type right?

File: week-2/Homework/mandatory/2-function-creation.js [hunk 8]
```diff
@@ -15,18 +25,30 @@ Complete the function to check if the variable `num` satisfies the following req
...
+Write a function that removes an element from an aray
 The function must:
 - NOT change the original array
 - return a new array with the item removed
 - remove the item at the specified index
 */
 
 function remove(arr, index) {
-  return; // complete this statement
+  arr.splice(index, 1);
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/2-function-creation.js:49]: So there is a mistake here. while your fix will pass, but it won't follow the condition (not change original array)
try to search splice destructive js
and try to do it using slice (twice maybe)

File: week-2/Homework/mandatory/2-function-creation.js [hunk 9]
```diff
@@ -36,8 +58,25 @@ Write a function that:
...
+  for (let j = 0; j < arr.length; j++) {
+    if (arr[j] > 100) {
+      arr[j] = 100;
+    }
+    newArray.push(parseFloat(arr[j].toFixed(2)) + "%")
+  }
+
+  return newArray
+
+}
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/2-function-creation.js:78]: Good one!

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 10]
```diff
@@ -12,13 +12,19 @@
...
+  //i= 0 , a = 7 , f2 = 4
+ i= 1 , a = 8 , f1 = 9
+ i= 2 , a = 9 , f2 = 6
+ i= 3 , a = 10 , f1 = 13
+ i= 4 , a = 11 , f2 = 8//
+
+  4. How many times is "f1" called?        //2//
+  5. How many times is "f2" called?        //3//
+ . What value does the "a" parameter take in the first "f1" call?       //1//
+  7. What is the value of the "a" outer variable when "f1" is called for the first time?    //8//
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/3-playing-computer.js:27]: all correct except 6 and 7
6. a will enter as 6+1+1=8
7. a will be 6 outside

File: week-2/Homework/mandatory/4-sorting-algorithm.js [hunk 11]
```diff
@@ -21,7 +21,25 @@ You don't have to worry about making this algorithm work fast! The idea is to ge
...
+  let arr1 = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (typeof arr[i] === "number") {
+      arr1.push(arr[i]);
+    }
+  }
+  for (i = 0; i < arr1.length; i++) {
+    for (j = i + 1; j < arr1.length; j++) {
+
+      if (arr1[i] > arr1[j]) {
```
Comment (Ashaghel) [inline week-2/Homework/mandatory/4-sorting-algorithm.js:33]: Bravo, interesting way to solve it

File: week-2/Homework/mandatory/1-fix-functions.js (hunk 4 above)
Comment (roza8765) [reply week-2/Homework/mandatory/1-fix-functions.js:57]: let sliced = numbers.slice(0, 5);
```suggestion

```

File: week-2/Homework/mandatory/1-fix-functions.js (hunk 5 above)
Comment (roza8765) [reply week-2/Homework/mandatory/1-fix-functions.js:70]: let element = arr[index];
```

File: week-2/Homework/mandatory/2-function-creation.js (hunk 7 above)
Comment (roza8765) [reply week-2/Homework/mandatory/2-function-creation.js:30]: (typeof num === "number" && num % 2 === 0 && num <= 100)
 where should we hve to usr typeof????????????

File: week-2/Homework/mandatory/2-function-creation.js (hunk 8 above)
Comment (roza8765) [reply week-2/Homework/mandatory/2-function-creation.js:49]: function remove(arr, index) {
  newArray = arr.slice(0);           //copying the arr
  newArray.splice(index, 1);
  console.log(newArray);
  return newArray;

File: week-2/Homework/mandatory/1-fix-functions.js (hunk 2 above)
Comment (roza8765) [reply week-2/Homework/mandatory/1-fix-functions.js:29]: const isBigEnough=num>=10;

File: week-2/Homework/mandatory/3-playing-computer.js (hunk 10 above)
Comment (roza8765) [reply week-2/Homework/mandatory/3-playing-computer.js:27]: I don't understand questions 6 and 7

**Pull request closed without merging**